src/common/misc.h

```diff
@@ -271,18 +271,13 @@ typedef struct {
 #endif
 #endif
 
-static INLINE CONST int isinff(float x) { return x == SLEEF_INFINITYf || x == -SLEEF_INFINITYf; }
-static INLINE CONST int isinfl(long double x) { return x == SLEEF_INFINITYl || x == -SLEEF_INFINITYl; }
-static INLINE CONST int isnanf(float x) { return x != x; }
-static INLINE CONST int isnanl(long double x) { return x != x; }
-
 #endif // defined(_MSC_VER)
 
-#if defined(__APPLE__) || defined(__MINGW32__) || defined(__MINGW64__)
-static INLINE CONST int isinff(float x) { return x == SLEEF_INFINITYf || x == -SLEEF_INFINITYf; }
-static INLINE CONST int isinfl(long double x) { return x == SLEEF_INFINITYl || x == -SLEEF_INFINITYl; }
-static INLINE CONST int isnanf(float x) { return x != x; }
-static INLINE CONST int isnanl(long double x) { return x != x; }
+#if defined(__APPLE__) || defined(__MINGW32__) || defined(__MINGW64__) || defined(__FreeBSD__) || defined(_MSC_VER)
```

Comment:
This look better now. One more question: can we define these macro unconditionally? Or will it break the build on linux because those are not defined? If linux is the only platform where these are already available, can we turn the `#if defined(...) || ... || defined(..)` into a `#ifndef __linux__`?

Reply:
I think it is not clean to define those macros unconditionally since they are semi-standardized.
I don't know if these macros are only defined in linux systems.